packages/platform-browser/src/dom/shared_styles_host.ts

```diff
@@ -36,18 +36,46 @@ export class SharedStylesHost {
 export class DomSharedStylesHost extends SharedStylesHost implements OnDestroy {
   // Maps all registered host nodes to a list of style nodes that have been added to the host node.
   private _hostNodes = new Map<Node, Node[]>();
+  private _styleNodesInDOM: Map<string|null, HTMLStyleElement>|undefined;
 
-  constructor(@Inject(DOCUMENT) private _doc: any) {
+  constructor(@Inject(DOCUMENT) private doc: Document, @Inject(APP_ID) private appId: string) {
     super();
-    this._hostNodes.set(_doc.head, []);
+
+    const styles: NodeListOf<HTMLStyleElement>|undefined =
+        this.doc.querySelectorAll(`style[ng-transition="${this.appId}"]`);
+
+    if (styles?.length) {
+      this._styleNodesInDOM = new Map(Array.from(styles).map((el) => [el.textContent, el]));
```

Comment:
nit: we can avoid extra loops and array allocations:
```suggestion
      const styleMap = new Map();
      for (const style of styles) {
        styleMap.set(style.textContent, style);
      }
      this._styleNodesInDOM = styleMap;
```